app/src/main/java/org/fossasia/pslab/communication/AnalyticsClass.java

```diff
@@ -0,0 +1,383 @@
+package org.fossasia.pslab.communication;
+
+import android.util.Log;
+
+import org.apache.commons.math3.analysis.ParametricUnivariateFunction;
+import org.apache.commons.math3.complex.Complex;
+import org.apache.commons.math3.fitting.CurveFitter;
+import org.apache.commons.math3.optim.nonlinear.vector.jacobian.LevenbergMarquardtOptimizer;
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+import org.apache.commons.math3.transform.DftNormalization;
+import org.apache.commons.math3.transform.FastFourierTransformer;
+import org.apache.commons.math3.transform.TransformType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.apache.commons.lang3.math.NumberUtils.min;
+import static org.apache.commons.lang3.math.NumberUtils.max;
+import static org.apache.commons.math3.util.FastMath.abs;
+import static org.apache.commons.math3.util.FastMath.exp;
+import static org.apache.commons.math3.util.FastMath.sin;
+
+/**
+ * Created by akarshan on 5/13/17.
+ * <p>
+ *     Unlike python, the curve fitting and the fourier transorm methods only accepts the arrays
+ *     of length of power of 2.
+ *
+ *     JAVA equivalent of scipy.fftpack.rfft, scipy.fftpack.rfftfreq and scipy.fftpack.fftfreq functions are'nt
+ *     available so they are been added in this class by the name fftToRfft, rfftFrequency and fftFrequency
+ *     respectively.
+ *
+ *      The fit functions have passed some dry tests but required to be tested with data points generated by PSLab.
+ * </p>
+ */
+
+@SuppressWarnings("ALL")
+public class AnalyticsClass {
+
+    // todo : check the accuracy of the all the curve fitting methods with data points generated by PSLab
+    //-------------------------- Exponential Fit ----------------------------------------//
+
+    ParametricUnivariateFunction exponentialParametricUnivariateFunction = new ParametricUnivariateFunction() {
+        @Override
+        public double value(double x, double... parameters) {
+            double a = parameters[0];
+            double b = parameters[1];
+            double c = parameters[2];
+            return a * exp(-x / b) + c;
+        }
+
+        @Override
+        public double[] gradient(double x, double... parameters) {
+            double a = parameters[0];
+            double b = parameters[1];
+            double c = parameters[2];
+
+            return new double[]{
+                    exp(-x / b),
+                    (a * exp(-x / b) * x) / b * b,
```

Comment:
Since you want b^2 in denominator, put b*b in brackets else it means divide & multiply by b.